app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package com.android.halanvendordiscovery.presentation.details.adapter
+
+import android.content.Intent
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat.startActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.android.halanvendordiscovery.databinding.ListItemMerchantBinding
+import com.android.halanvendordiscovery.domain.details.model.MerchantDomainModel
+
+class MerchantsAdapter(
+    private val list: List<MerchantDomainModel>
+) : RecyclerView.Adapter<MerchantsAdapter.ViewHolder>() {
+    inner class ViewHolder(private val binding: ListItemMerchantBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(item: MerchantDomainModel) {
+            binding.tvMerchantName.text = item.arabicName
+            binding.tvMerchantAddress.text = item.address
+            binding.ivPhone.setOnClickListener {
+                val uri = "tel:${item.phoneNumber}"
+                val intent = Intent(Intent.ACTION_VIEW, Uri.parse(uri))
+                startActivity(binding.root.context, intent, null)
```

Comment:
you can add this logic in fragment or activity which contain adapter by listener

Reply:
Confirmed 🚀